### ISSUE TITLE
expectRevert does not work

### ISSUE THREAD

**Issue by cacampbell** [EthWorks/Waffle#393]
Contract:
```
function call() external payable returns (bool) {
        require(
            false, 
            "Incorrect fee amount, call getAccountCreationFee"
        );
}
```

Test:
```
    it("reverts", async () => {
        await expect(contract.connect(accounts[1]).call({
            value: ethers.utils.parseEther("1")
        })).to.be.reverted;
    });
```

Result:
```
AssertionError: Expected transaction to be reverted
      + expected - actual

      -Transaction NOT reverted.
      +Transaction reverted.
```

```
❯ cat yarn.lock | grep waffle
"@ethereum-waffle/chai@^3.2.0":
  resolved "https://registry.yarnpkg.com/@ethereum-waffle/chai/-/chai-3.2.0.tgz#f75465e447358db0fde967b26cf10adbe5365380"
    "@ethereum-waffle/provider" "^3.2.0"
"@ethereum-waffle/compiler@^3.2.0":
  resolved "https://registry.yarnpkg.com/@ethereum-waffle/compiler/-/compiler-3.2.0.tgz#3d5cccc8171ff61a0069d9558fb567ddd45b6b4e"
"@ethereum-waffle/ens@^3.2.0":
  resolved "https://registry.yarnpkg.com/@ethereum-waffle/ens/-/ens-3.2.0.tgz#26f7b2048a422eae5ce39a855e519e01ebc06953"
"@ethereum-waffle/mock-contract@^3.2.0":
  resolved "https://registry.yarnpkg.com/@ethereum-waffle/mock-contract/-/mock-contract-3.2.0.tgz#be9441060f8b58f1d7e344f74e5dd158d41a7c45"
"@ethereum-waffle/provider@^3.2.0":
  resolved "https://registry.yarnpkg.com/@ethereum-waffle/provider/-/provider-3.2.0.tgz#501d4bbf04ffc01dced0107dcc9e20f24354387f"
    "@ethereum-waffle/ens" "^3.2.0"
"@nomiclabs/buidler-waffle@^2.1.0":
  resolved "https://registry.yarnpkg.com/@nomiclabs/buidler-waffle/-/buidler-waffle-2.1.0.tgz#9869ac829162eb57ac4e4096f0a02bd7ef54a71a"
"@nomiclabs/hardhat-waffle@^2.0.0":
  resolved "https://registry.yarnpkg.com/@nomiclabs/hardhat-waffle/-/hardhat-waffle-2.0.0.tgz#b06533c802da1a9d614e432f719d2816384df9b3"
ethereum-waffle@^3.1.2:
  resolved "https://registry.yarnpkg.com/ethereum-waffle/-/ethereum-waffle-3.2.0.tgz#1cea66fece731ea281e9d68046056caed2f2af84"
    "@ethereum-waffle/chai" "^3.2.0"
    "@ethereum-waffle/compiler" "^3.2.0"
    "@ethereum-waffle/mock-contract" "^3.2.0"
    "@ethereum-waffle/provider" "^3.2.0"
```

**Comment by cacampbell** [comment]
Hi, also of note:

This is only the case when I run this using the in memory hardhat evm. 

If I run the hardhat evm separately and then run tests with `--network localhost`, the revert is picked up correctly

**Comment by marekkirejczyk** [comment]
Fixed by #442 and released in 3.3.0
